text/0000-placement-left-arrow.md

```diff
@@ -0,0 +1,164 @@
+the `in` keyword, instead use the syntax `PLACE_EXPR <- VALUE_EXPR`.
+
+This takes advantage of the fact that `<-` was reserved as a token via
+historical accident (that for once worked out in our favor).
+
+# Motivation
+
+One sentence: the syntax `a <- b` is short, can be parsed without
+ambiguity, and is strongly connotated already with assignment.
+
+Further text (essentially historical background):
+
+There is much debate about what syntax to use for placement-new.
+We started with `box (PLACE_EXPR) VALUE_EXPR`, then migrated towards
+leveraging the `in` keyword instead of `box`, yielding `in (PLACE_EXPR) VALUE_EXPR`.
+
+A lot of people disliked the `in (PLACE_EXPR) VALUE_EXPR` syntax
+(see discussion from [RFC 809]).
+
+[RFC 809]: https://github.com/rust-lang/rfcs/pull/809
+
+In response to that discussion (and also due to personal preference)
+I suggested the alternative syntax `in PLACE_EXPR { BLOCK_EXPR }`,
+which is what landed when [RFC 809] was merged.
+
+However, it is worth noting that this alternative syntax actually
+failed to address a number of objections (some of which also
+applied to the original `in (PLACE_EXPR) VALUE_EXPR` syntax):
+
+ * [kennytm](https://github.com/rust-lang/rfcs/pull/809#issuecomment-73071324)
+
+   > While in (place) value is syntactically unambiguous, it looks
+   > completely unnatural as a statement alone, mainly because there
+   > are no verbs in the correct place, and also using in alone is
+   > usually associated with iteration (for x in y) and member
+   > testing (elem in set).
+
+ * [petrochenkov](https://github.com/rust-lang/rfcs/pull/809#issuecomment-73142168)
+
+   > As C++11 experience has shown, when it's available, it will
+   > become the default method of inserting elements in containers,
+   > since it's never performing worse than "normal insertion" and
+   > is often better. So it should really have as short and
+   > convenient syntax as possible.
+
+ * [p1start](https://github.com/rust-lang/rfcs/pull/809#issuecomment-73837430)
+
+   > I’m not a fan of in <place> { <stmts> }, simply because the
+   > requirement of a block suggests that it’s some kind of control
+   > flow structure, or that all the statements inside will be
+   > somehow run ‘in’ the given <place> (or perhaps, as @m13253
+   > seems to have interpreted it, for all box expressions to go
+   > into the given place). It would be our first syntactical
+   > construct which is basically just an operator that has to
+   > have a block operand.
+
+I believe the `PLACE_EXPR <- VALUE_EXPR` syntax addresses all of the
+above concerns.
+
+Thus cases like allocating into an arena (which needs to take as input the arena itself
+and a value-expression, and returns a reference or handle for the allocated entry in the arena -- i.e. *cannot* return unit)
+would look like:
+
+```rust
+let ref_1 = arena <- value_expression;
+let ref_2 = arena <- value_expression;
```

Comment:
@nagisa That wouldn’t really work for all cases, because, e.g., `[Typed]Arena::alloc` takes `&self`, not `&mut self`, meaning that you’d be forced to use mutable references when you shouldn’t have to.